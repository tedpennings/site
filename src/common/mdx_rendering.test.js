import { renderWithReact } from "@mdx-js/test-util";
import { screen } from "@testing-library/dom";
import "@testing-library/jest-dom/extend-expect";

import mdxComponents from "./mdx_components";

async function renderMarkdown(markdown) {
  const html = await renderWithReact(markdown, { components: mdxComponents });
  document.body.innerHTML = html;
  return html;
}

describe("MDX rendering", () => {
  test.each([1, 2, 3, 4, 5, 6])(`renders h%i tags correctly`, async (level) => {
    await renderMarkdown(headingMarkdown(level));
    expect(screen.getByRole("heading", { level })).toHaveTextContent(
      `heading ${headingWords[level]}`
    );
    // Mui class is important for styling
    expect(screen.getByRole("heading", { level })).toHaveClass(
      `MuiTypography-h${level}`
    );
  });

  it("renders a paragraph with the appropriate styling", async () => {
    const html = await renderMarkdown(paragraph);
    expect(html).toMatchSnapshot();
  });
  it("renders lists with the appropriate styling", async () => {
    const html = await renderMarkdown(lists);
    expect(html).toMatchSnapshot();
  });
});

const headingWords = [null, "one", "two", "three", "four", "five", "six"];
const hashes = (count) => new Array(count).fill("#").join("");
const headingMarkdown = (level) =>
  [hashes(level), "heading", headingWords[level], "\n\n"].join(" ");

const paragraph = `
"But do you believe," said Candide, "that the earth was originally a sea, as we find it asserted in that large book belonging to the captain?"

"I do not believe a word of it," said Martin, "any more than I do of the many ravings which have been published lately."

"But for what end, then, has this world been formed?" said Candide.

"To plague us to death," answered Martin.
`;

const lists = `
In 2017, Jen I and I traveled to:

1. New Zealand
1. Australia
1. Indonesia
1. Taiwan
1. Malaysia
1. Thailand
1. India
1. Singapore
1. UAE
1. Egypt
1. Morocco
1. Spain

We brought with us:
* Backpacks
* Love
* Mini Bose speaker
* Warm weather clothes, which wasn't right for the Himalayas
`;
