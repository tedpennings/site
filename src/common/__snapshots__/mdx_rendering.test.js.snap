// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MDX rendering renders a paragraph with the appropriate styling 1`] = `"<p class=\\"MuiTypography-root MuiTypography-body1 MuiTypography-paragraph\\">&quot;But do you believe,&quot; said Candide, &quot;that the earth was originally a sea, as we find it asserted in that large book belonging to the captain?&quot;</p><p class=\\"MuiTypography-root MuiTypography-body1 MuiTypography-paragraph\\">&quot;I do not believe a word of it,&quot; said Martin, &quot;any more than I do of the many ravings which have been published lately.&quot;</p><p class=\\"MuiTypography-root MuiTypography-body1 MuiTypography-paragraph\\">&quot;But for what end, then, has this world been formed?&quot; said Candide.</p><p class=\\"MuiTypography-root MuiTypography-body1 MuiTypography-paragraph\\">&quot;To plague us to death,&quot; answered Martin.</p>"`;

exports[`MDX rendering renders lists with the appropriate styling 1`] = `"<p class=\\"MuiTypography-root MuiTypography-body1 MuiTypography-paragraph\\">In 2017, Jen I and I traveled to:</p><ol class=\\"MuiBox-root MuiBox-root-7\\" style=\\"padding-inline-start:0\\"><li class=\\"MuiBox-root MuiBox-root-8\\"><svg class=\\"MuiSvgIcon-root MuiSvgIcon-colorSecondary\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><path d=\\"M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z\\"></path></svg>New Zealand</li><li class=\\"MuiBox-root MuiBox-root-9\\"><svg class=\\"MuiSvgIcon-root MuiSvgIcon-colorSecondary\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><path d=\\"M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z\\"></path></svg>Australia</li><li class=\\"MuiBox-root MuiBox-root-10\\"><svg class=\\"MuiSvgIcon-root MuiSvgIcon-colorSecondary\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><path d=\\"M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z\\"></path></svg>Indonesia</li><li class=\\"MuiBox-root MuiBox-root-11\\"><svg class=\\"MuiSvgIcon-root MuiSvgIcon-colorSecondary\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><path d=\\"M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z\\"></path></svg>Taiwan</li><li class=\\"MuiBox-root MuiBox-root-12\\"><svg class=\\"MuiSvgIcon-root MuiSvgIcon-colorSecondary\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><path d=\\"M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z\\"></path></svg>Malaysia</li><li class=\\"MuiBox-root MuiBox-root-13\\"><svg class=\\"MuiSvgIcon-root MuiSvgIcon-colorSecondary\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><path d=\\"M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z\\"></path></svg>Thailand</li><li class=\\"MuiBox-root MuiBox-root-14\\"><svg class=\\"MuiSvgIcon-root MuiSvgIcon-colorSecondary\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><path d=\\"M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z\\"></path></svg>India</li><li class=\\"MuiBox-root MuiBox-root-15\\"><svg class=\\"MuiSvgIcon-root MuiSvgIcon-colorSecondary\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><path d=\\"M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z\\"></path></svg>Singapore</li><li class=\\"MuiBox-root MuiBox-root-16\\"><svg class=\\"MuiSvgIcon-root MuiSvgIcon-colorSecondary\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><path d=\\"M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z\\"></path></svg>UAE</li><li class=\\"MuiBox-root MuiBox-root-17\\"><svg class=\\"MuiSvgIcon-root MuiSvgIcon-colorSecondary\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><path d=\\"M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z\\"></path></svg>Egypt</li><li class=\\"MuiBox-root MuiBox-root-18\\"><svg class=\\"MuiSvgIcon-root MuiSvgIcon-colorSecondary\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><path d=\\"M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z\\"></path></svg>Morocco</li><li class=\\"MuiBox-root MuiBox-root-19\\"><svg class=\\"MuiSvgIcon-root MuiSvgIcon-colorSecondary\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><path d=\\"M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z\\"></path></svg>Spain</li></ol><p class=\\"MuiTypography-root MuiTypography-body1 MuiTypography-paragraph\\">We brought with us:</p><ul class=\\"MuiBox-root MuiBox-root-20\\" style=\\"padding-inline-start:0\\"><li class=\\"MuiBox-root MuiBox-root-21\\"><svg class=\\"MuiSvgIcon-root MuiSvgIcon-colorSecondary\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><path d=\\"M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z\\"></path></svg>Backpacks</li><li class=\\"MuiBox-root MuiBox-root-22\\"><svg class=\\"MuiSvgIcon-root MuiSvgIcon-colorSecondary\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><path d=\\"M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z\\"></path></svg>Love</li><li class=\\"MuiBox-root MuiBox-root-23\\"><svg class=\\"MuiSvgIcon-root MuiSvgIcon-colorSecondary\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><path d=\\"M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z\\"></path></svg>Mini Bose speaker</li><li class=\\"MuiBox-root MuiBox-root-24\\"><svg class=\\"MuiSvgIcon-root MuiSvgIcon-colorSecondary\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><path d=\\"M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z\\"></path></svg>Warm weather clothes, which wasn&#x27;t right for the Himalayas</li></ul>"`;
